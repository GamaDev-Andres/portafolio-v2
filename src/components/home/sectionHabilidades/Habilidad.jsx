import propTypes from 'prop-types'

const Habilidad = ({ porcentaje, children }) => {
  return (
    <div>
      <span className="font-medium">{children}</span>
      <div className="bg-fondos w-full  flex  rounded-full">
        <div className={`bg-letra  ${porcentaje} rounded-full p-1`}>
          <span className="text-white w-full text-center block ">
            {porcentaje.slice(3, 5)} %
          </span>
        </div>
      </div>
    </div>
  )
}
Habilidad.propTypes = {
  porcentaje: propTypes.string.isRequired,
  children: propTypes.string.isRequired
}
export default Habilidad
