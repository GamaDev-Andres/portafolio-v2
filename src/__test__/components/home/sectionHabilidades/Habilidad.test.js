/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import Habilidad from '../../../../components/home/sectionHabilidades/Habilidad'

describe('probando <Habilidad/>', () => {
  const children = 'Desarrollo front-end'
  const wrapper = Enzyme.mount(
    <Habilidad porcentaje="w-[80%]">{children}</Habilidad>

  )
  test('debe montarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('span').at(0).text().trim()).toBe(children)
  })
})
