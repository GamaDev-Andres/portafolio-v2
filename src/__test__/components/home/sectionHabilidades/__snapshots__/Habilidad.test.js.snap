// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <Habilidad/> debe montarse correctamente 1`] = `
<div>
  <span
    className="font-medium"
  >
    Desarrollo front-end
  </span>
  <div
    className="bg-fondos w-full  flex  rounded-full"
  >
    <div
      className="bg-letra  w-[80%] rounded-full p-1"
    >
      <span
        className="text-white w-full text-center block "
      >
        80
         %
      </span>
    </div>
  </div>
</div>
`;
